### PR TITLE
Add *Tags* index page

### PR DESCRIPTION
**Pull request by straight-shoota**
This is a follow-up to https://github.com/crystal-lang/crystal-website/pull/711 adding an index for all the tags.
Previously, tags were only shown on each post and there are individual pages for each tag showing the tagged posts. This patch adds an overview page listing _all_ tags.
This is useful even just as an internal editing tool 😛 

Tags with most articles are ordered on top.
The most recent three articles per tag are rendered, and a link to the tag page.

grafik

### CODE REVIEW

File: _pages/blog/tags.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+title: Tags
+layout: page
+section: tags
+---
+
+{% capture counts_with_tags_string %}{% for tag in site.tags %}{{ tag[1] | size | prepend:"000000" | slice:-6,6 }}:{{ tag[0] }}{% unless forloop.last %},{% endunless %}{% endfor %}{% endcapture %}
```
Comment (beta-ziliani) [inline _pages/blog/tags.md:7]: really there isn't a better way to format a number?? 😮

**Review by beta-ziliani — APPROVED** [review]
Nice!

File: _pages/blog/tags.md (hunk 1 above)
Comment (straight-shoota) [reply _pages/blog/tags.md:7]: Nope. Apparently not 🤷

Comment (ysbaddaden) [reply _pages/blog/tags.md:7]: We probably can, but that would need to write a template helper function.

Comment (straight-shoota) [reply _pages/blog/tags.md:7]: Yes, of course with a plugin we could make this cleaner.

**Pull request merged**